packages/manager/modules/sms/src/sms/options/smppParameter/smppParameter.controller.js

```diff
@@ -1,19 +1,44 @@
 export default class SmsSmppParameterCtrl {
   /* @ngInject */
-  constructor(SmsService) {
+  constructor(SmsService, $timeout, $state) {
     this.smsService = SmsService;
+    this.$timeout = $timeout;
+    this.$state = $state;
   }
 
   $onInit() {
+    this.checkStatus = this.checkStatus.bind(this);
     this.OPTION_SMPP_TRACKING_PREFIX = 'sms::service::configure-smpp::';
     this.loading = true;
     return this.smsService
       .getSmppSettings(this.serviceName)
       .then((result) => {
         this.smppSettings = result;
+        if (this.smppSettings.status === 'UPDATING_IP') {
+          this.checkStatus();
+        }
       })
       .finally(() => {
         this.loading = false;
       });
   }
+
+  $onDestroy() {
+    this.checkStatus = () => false;
+  }
+
+  checkStatus() {
+    return this.smsService
+      .getSmppSettings(this.serviceName)
+      .then(({ status }) => {
+        this.smppSettings.status = status;
+        if (
+          status === 'UPDATING_IP' &&
+          this.$state.current.name === 'sms.service.options.smppParameter'
+        ) {
+          return this.$timeout(() => this.checkStatus(), 5000);
```

Comment:
**nitpick:** `$interval` is appropriate for this use-case. instead of `$timeout`. Could you please change the implementation?
Let me know if you need any information.

Reply:
I don't have the same expected result.